ticket_migrate.py

```diff
@@ -0,0 +1,829 @@
+
+    for issue in issues:
+        issue.submitMyComments(comments)
+        # Submit to project before closing the issue,
+        # so the issue may auto-move to another column (such as "Done").
+        issue.submitToProject()
+        issue.closeIfNeeded()
+
+
+def select_tickets(tickets):
+    """
+    Easy-to-edit method to choose tickets to submit.
+    Useful in case of failure while creating tickets.
+    You can check here that the `t_id` is not in `tickets_created.tsv`.
+    """
+    # return [t for t in tickets if t['t_id'] == 5519]
+    # return [t for t in tickets if t['status'] != 'closed']
+    return tickets
+
+
+def read_trac_tickets():
+    """
+    Read the Trac ticket data from the database, and generate dicts.
+    """
+    db = get_db()
+
+    # Only take the last branch change.
+    # For example, https://trac.chevah.com/ticket/85 has multiple changes,
+    # and the last one is to GH PR 238.
+    # We use GROUP BY because SQLite has no DISTINCT ON.
+    # https://www.sqlite.org/quirks.html#aggregate_queries_can_contain_non_aggregate_result_columns_that_are_not_in_the_group_by_clause
+    for row in db.execute(
+            """\
+            SELECT *
+            FROM
+              (SELECT *
+               FROM ticket
+               LEFT JOIN ticket_change ON ticket.id = ticket_change.ticket
+               AND ticket_change.field = 'branch'
+               AND ticket_change.newvalue LIKE '%github%'
+               ORDER BY ticket.id,
+                        ticket_change.time DESC)
+            GROUP BY id;
+            """):
+        (
+            t_id,
+            t_type,
+            time,
+            changetime,
+            component,
+            severity,
+            priority,
+            owner,
+            reporter,
+            cc,
+            version,
+            milestone,
+            status,
+            resolution,
+            summary,
+            description,
+            keywords,
+            _ticket,
+            _time,
+            _author,
+            _field,
+            _oldvalue,
+            _newvalue,
+            ) = row
+
+        yield {
+            't_id': t_id,
+            't_type': t_type,
+            'time': time,
+            'changetime': changetime,
+            'component': component,
+            'severity': severity,
+            'priority': priority,
+            'owner': owner,
+            'reporter': reporter,
+            'cc': cc,
+            'version': version,
+            'milestone': milestone,
+            'status': status,
+            'resolution': resolution,
+            'summary': summary,
+            'description': description,
+            'keywords': keywords,
+            'branch': _newvalue,
+            }
+
+
+def read_trac_comments():
+    """
+    Read the Trac comment data from the database.
+
+    The last version is in the `newvalue` of the `comment` field.
+
+    Example changed comment: https://trac.chevah.com/ticket/3928#comment:1
```

Comment:
this is open source code ...but trac.chevah.com is private ...so these links are not very helpful 

we can leave them... no problem....

but we also plan to decommission trac.chevah.com so the links will be broken

and I don't know if we will setup redirection...but we might :)